jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/Dispatcher.java

```diff
@@ -259,6 +259,12 @@ public String getRequestURI()
             return _uri == null ? super.getRequestURI() : _uri.getPath();
         }
 
+        @Override
+        public StringBuffer getRequestURL()
+        {
+            return _uri == null ? super.getRequestURL() :  new StringBuffer(HttpURI.build(_uri).query(null).scheme(super.getScheme()).host(super.getServerName()).port(super.getServerPort()).asString());
```

Comment:
Should this ever  be returning `super.getRequestURL()`?

From reading https://github.com/jakartaee/servlet/issues/309 and getRequestURL javadoc.

> If this request has been forwarded using RequestDispatcher.forward, the server path in the reconstructed URL must reflect the path used to obtain the RequestDispatcher, and not the server path specified by the client.

It seems `getRequestURL` and `getRequestURI` should never be returning values for the source value but only for the target of the current dispatch.

Reply:
@lachlan-roberts named dispatcher don't have paths